### DISCUSSION TITLE
Rewrite api routes to an api.* subdomain for public use

### DISCUSSION THREAD

**Discussion by cjroebuck** [blitz-js/blitz#2750] [Q&A]
Hi there, 

I’m currently evaluating using blitz and just wondering if it’s possible to use the API routes externally somehow. From the docs it looks like this is currently one of the tradeoffs of adapting blitz.

I imagine using blitz to make the ‘dashboard’ e.g. myapp.com webapp, but I’d need to reuse some of the backend api routes and place them at api.myapp.com so that they can be consumed by any api client (without session/csrf cookie etc and just using api key)

If this isn’t possible or currently recommended, what would you advise to be the best way to create public api endpoints but trying to reuse code from the blitz api route, such as validation logic/models/db access etc

**Comment by flybayer** [comment]
Hey, sorry for the delay in responding!

Are you wanting to use the auto generated API routes externally? If so, you can definitely do that, we have people doing that in production. But we don't yet have first class support for using an api key. (related: https://github.com/blitz-js/blitz/issues/2614).

Otherwise you can add [custom API routes](https://blitzjs.com/docs/api-routes) where you can do anything you want, including re-using server code from queries/mutations.

And I think you should be able to do your url rewrite with [rewrites](https://blitzjs.com/docs/rewrites).

**Reply by Jarrodsz** [reply]
If understand correctly one would do something like:

```module.exports = {
  async rewrites() {
    return [
      {
        source: "/blog/:slug",
        destination: "https://api.mydomain.ltd/blog/:slug", // Matched parameters can be used in the destination
      },
    ]
  },
}


That is all that is required?

**Reply by flybayer** [reply]
I think so!
